### PR TITLE
Handle guard when OSPlatform.Create used, fix typo

### PR DESCRIPTION
**Pull request by buyaa-n**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/4119 plus fixed typo `PlatformCompatabilityAnalyzer => PlatformCompatibilityAnalyzer` which found everyhere

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs [hunk 1]
```diff
@@ -128,6 +128,18 @@ private static bool TryExtractPlatformName(string methodName, [NotNullWhen(true)
...
                     return true;
                 }
+                else if (argumentValue is IInvocationOperation iOperaion &&
+                    iOperaion.TargetMethod.ReturnType.Equals(osPlatformType) &&
+                    iOperaion.Arguments.Length == 1 &&
+                    iOperaion.Arguments[0] is { } argument &&
+                    argument.Value is ILiteralOperation literal &&
+                    literal.Type.SpecialType == SpecialType.System_String &&
+                    literal.ConstantValue.HasValue &&
+                    !literal.ConstantValue.Value.Equals(string.Empty))
```
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs:138]: Wouldn't it be better to check this is a string?

```suggestion
                    literal.ConstantValue.Value is string platformName &&
                    platformName.Length > 0)
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs [hunk 2]
```diff
@@ -128,6 +128,18 @@ private static bool TryExtractPlatformName(string methodName, [NotNullWhen(true)
...
+                {
+                    osPlatformName = literal.ConstantValue.Value.ToString();
```
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs:140]: If you are going for the previous change, you could simply do

```suggestion
                    osPlatformName = platformName;
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs [hunk 3]
```diff
@@ -128,6 +128,18 @@ private static bool TryExtractPlatformName(string methodName, [NotNullWhen(true)
                     osPlatformName = propertyReference.Property.Name;
                     return true;
                 }
+                else if (argumentValue is IInvocationOperation iOperaion &&
```
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs:131]: You could optionally extract this big if condition as a method.

**Comment by buyaa-n** [comment]
Thanks, @Evangelink for review, @mavasani resolved this with his latest PR, i was keeping for renaming changes, but from the number of conflicts on this branch seems its better to close it

**Pull request closed without merging**